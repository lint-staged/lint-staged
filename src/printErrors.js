/* eslint no-console: 0 */

'use strict'

module.exports = function printErrors(errorInstance) {
  if (Array.isArray(errorInstance.errors)) {
    errorInstance.errors.forEach(lintError => {
      console.error(lintError.message)
    })
  } else {
    console.error(errorInstance.message)
  }
}
