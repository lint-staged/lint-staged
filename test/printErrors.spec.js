/* eslint no-unused-expressions: 0 */
/* eslint no-console: 0 */
/* eslint no-global-assign: 0 */

import Listr from 'listr'
import { makeConsoleMock } from 'consolemock'
import printErrors from '../src/printErrors'

describe('printErrors', () => {
  beforeEach(() => {
    console = makeConsoleMock()
  })
  it('should print plain errors', () => {
    const err = new Error('We have a problem')
    printErrors(err)
    expect(console.printHistory()).toMatchSnapshot()
  })

  it('should print Listr nested errors', async () => {
    const list = new Listr(
      [
        {
          title: 'foo',
          task: () => Promise.reject(new Error('Foo failed'))
        },
        {
          title: 'bar',
          task: () =>
            new Listr([
              {
                title: 'unicorn',
                task: () => Promise.reject(new Error('Unicorn failed'))
              },
              {
                title: 'rainbow',
                task: () => Promise.resolve()
              }
            ])
        },
        {
          title: 'baz',
          task: () => Promise.resolve()
        }
      ],
      {
        exitOnError: false,
        renderer: 'silent'
      }
    )

    try {
      await list.run()
    } catch (err) {
      printErrors(err)
      expect(console.printHistory()).toMatchSnapshot()
    }
  })
})
