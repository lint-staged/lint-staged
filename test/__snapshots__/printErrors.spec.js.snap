// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`printErrors should print Listr nested errors 1`] = `
"
ERROR Foo failed
ERROR Unicorn failed"
`;

exports[`printErrors should print plain errors 1`] = `
"
ERROR We have a problem"
`;
